File: Graph/C#/1389 케빈 베이컨의 6단계 법칙.cs
using System;
using System.Collections.Generic;
using static System.Console;


class Program
{
    static List<List<int>> graph = new List<List<int>>();
    static int[] relate = new int[101];
    static int[] friend = new int[101];
    static int n, m;
    
    static void bfs(int start) {
        Queue<int> queue = new Queue<int>();
        relate[start] = 0;
        queue.Enqueue(start);
        
        while(queue.Count != 0) {
            int v = queue.Peek(); queue.Dequeue();
            foreach(int to in graph[v]) {
                if(relate[to] == -1) {
                    relate[to] = relate[v]+1;
                    queue.Enqueue(to);
                }
            }
        }
    }
    
    static void Init() {
        Array.Fill(relate, -1, 0, 101);
    }
    
    static void Main() {
        int answer = 0;
        int min_value = 10001;
        string[] s = ReadLine().Split(' ');
        n = int.Parse(s[0]); m = int.Parse(s[1]);
        for(int i = 0 ; i < n+1; i++) {
            graph.Add(new List<int>());
        }
        for(int i = 0 ; i < m; i++) {
            string[] fromto = ReadLine().Split(' ');
            int from = int.Parse(fromto[0]); int to = int.Parse(fromto[1]);
            graph[from].Add(to); graph[to].Add(from);
        }
        
        for(int i = 1; i <= n; i++) {
            Init();
            bfs(i);
            int c = 0;
            for(int j = 1; j <= n; j++) {
                c += relate[j];
            }
            min_value = Math.Min(min_value, c);
            friend[i] = c;
        }
        
        
        answer = Array.IndexOf(friend, min_value);
        Write(answer);
    }
}

File: Graph/Python/1389 케빈 베이컨의 6단계 법칙.py
from collections import deque

answer = []
n,m = map(int, input().split())
graph = [[] for i in range(n+1)]
visited = [-1 for i in range(n+1)]

for i in range(m) :
    f,t = map(int, input().split())
    graph[f].append(t)
    graph[t].append(f)
    
def bfs(start) :
    queue = deque()
    queue.append(start)
    visited[start] = 0
    while queue :
        v = queue.popleft()
        for to in graph[v] :
            if visited[to] == -1 :
                visited[to] = visited[v] + 1
                queue.append(to)

for i in range(1, n+1) :
    bfs(i)
    answer.append(sum(visited[1:n+1]))
    visited = [-1 for i in range(n+1)]

print(answer.index(min(answer))+1)
